src/Microsoft.Diagnostics.Monitoring.EventPipe/Traces/TraceEventExtensions.cs

```diff
@@ -0,0 +1,220 @@
+                        case "ActivityTraceFlags":
+                            if (value is string traceFlagsValue)
+                            {
+                                traceFlags = (ActivityTraceFlags)Enum.Parse(typeof(ActivityTraceFlags), traceFlagsValue);
+                            }
+                            break;
+                        case "TraceStateString":
+                            traceState = value as string;
+                            break;
+                        case "Kind":
+                            if (value is string kindValue)
+                            {
+                                kind = (ActivityKind)Enum.Parse(typeof(ActivityKind), kindValue);
+                            }
+                            break;
+                        case "DisplayName":
+                            string? displayNameValue = value as string;
+                            if (!string.IsNullOrEmpty(displayNameValue)
+                                && displayNameValue != activityName)
+                            {
+                                displayName = displayNameValue;
+                            }
+
+                            break;
+                        case "StartTimeTicks":
+                            if (value is string startTimeUtcValue)
+                            {
+                                startTimeUtc = DateTime.SpecifyKind(new DateTime(long.Parse(startTimeUtcValue)), DateTimeKind.Utc);
+                            }
+                            break;
+                        case "DurationTicks":
+                            if (value is string durationTicksValue)
+                            {
+                                durationTicks = long.Parse(durationTicksValue);
+                            }
+                            break;
+                        case "Status":
+                            if (value is string statusValue)
+                            {
+                                status = (ActivityStatusCode)Enum.Parse(typeof(ActivityStatusCode), statusValue);
+                            }
+                            break;
+                        case "StatusDescription":
+                            statusDescription = value as string;
+                            break;
+                        case "Tags":
+                            string? tagsValue = value as string;
+                            if (!string.IsNullOrEmpty(tagsValue))
+                            {
+                                tagCount = ParseTags(tagsValue);
+                            }
+                            break;
+                        case "ActivitySourceVersion":
+                            sourceVersion = value as string;
+                            break;
+                    }
+                }
+
+                if (!s_Sources.TryGetValue(sourceName, out source))
+                {
+                    source = new(sourceName, sourceVersion);
```

Comment:
So what happens if there are two events that have the same source name but different versions?

Reply:
We'll just report the version of the first one we saw. It is kind of an edge case thing. I doubt multiple sources with different versions really happens in practice (it could) but even if it did I don't think there would be big impact. We could make the key a tuple if you are concerned. Or we could do it later if anyone raises an issue?